packages/shylu/shylu_node/fastilu/src/shylu_fastilu.hpp

```diff
@@ -2327,12 +2567,96 @@ class FastILUFunctor
                         _Ux[uptr-1] = ((one - _omega) * _Ux[uptr - 1]) + (_omega * val);
                     }
                 }
+        }
+
+        KOKKOS_INLINE_FUNCTION
+        void functor_bcrs_impl(const Ordinal start, const Ordinal end) const
```

Comment:
This functor was, by far, the biggest struggle. Getting something that matched the exact behavior of the unblocked impl was tricky. I tried to keep it as similar to the unblocked impl as possible, not sure if a different impl that took advantage of blocks more would be better.

Reply:
I would have gone with the later, you went with the former, that is ok. At some point we might want to implement the later and compare.